File: Python/test_roman_numerals.py
# -*- encoding: utf-8 -*-
# Test Suite for Roman Numeral Converter
# Copyright © MMXVI-MMXVII, Chris Warrick.
# All rights reserved.
# License: 3-clause BSD, see main file for details.

"""Test suite for Roman Numeral Converter."""
import pytest
from roman_numerals import to_roman, from_roman


def test_to_roman():
    assert to_roman(1) == 'I'
    assert to_roman(2) == 'II'
    assert to_roman(3) == 'III'
    assert to_roman(4) == 'IV'
    assert to_roman(5) == 'V'
    assert to_roman(6) == 'VI'
    assert to_roman(1234) == 'MCCXXXIV'
    assert to_roman(1958) == 'MCMLVIII'
    assert to_roman(2222) == 'MMCCXXII'
    assert to_roman(3999) == 'MMMCMXCIX'


def test_to_roman_invalid():
    with pytest.raises(ValueError):
        to_roman(0)
    with pytest.raises(ValueError):
        to_roman(-1)
    with pytest.raises(ValueError):
        to_roman(4000)


def test_to_roman_file():
    with open('../test_data.txt') as fh:
        for line in fh:
            integer, roman = line.strip().split(' ')
            integer = int(integer)
            assert to_roman(integer) == roman


def test_from_roman():
    assert from_roman("I") == 1
    assert from_roman("Ii") == 2
    assert from_roman("iii") == 3
    assert from_roman("  iv  ") == 4
    assert from_roman("V") == 5
    assert from_roman("ViIi") == 8


def test_from_roman_file():
    with open('../test_data.txt') as fh:
        for line in fh:
            integer, roman = line.strip().split(' ')
            integer = int(integer)
            assert from_roman(roman) == integer


if __name__ == '__main__':
    print("Please run with py.test.")

File: Ruby/test_roman_numerals.rb
# -*- encoding: utf-8 -*-
# Test Suite for Roman Numeral Converter
# Copyright © MMXVI-MMXVII, Chris Warrick.
# All rights reserved.
# License: 3-clause BSD, see main file for details.

require "./roman_numerals"
require "test/unit"

class TestRomanNumerals < Test::Unit::TestCase
    def test_to_roman
        assert_equal("I", to_roman(1))
        assert_equal("II", to_roman(2))
        assert_equal("III", to_roman(3))
        assert_equal("IV", to_roman(4))
        assert_equal("V", to_roman(5))
        assert_equal("VI", to_roman(6))
        assert_equal("MCCXXXIV", to_roman(1234))
        assert_equal("MCMLVIII", to_roman(1958))
        assert_equal("MMCCXXII", to_roman(2222))
        assert_equal("MMMCMXCIX", to_roman(3999))
    end


    def test_to_roman_invalid
        assert_raise(RangeError) { to_roman(0) }
        assert_raise(RangeError) { to_roman(-1) }
        assert_raise(RangeError) { to_roman(4000) }
    end

    def test_to_roman_file
        file = IO.readlines("../test_data.txt")
        file.each do |line|
            integer, roman = line.strip.split(" ")
            integer = integer.to_i
            assert_equal(roman, to_roman(integer))
        end
    end


    def test_from_roman
        assert_equal(1, from_roman("I"))
        assert_equal(2, from_roman("Ii"))
        assert_equal(3, from_roman("iii"))
        assert_equal(4, from_roman("  iv  "))
        assert_equal(5, from_roman("V"))
        assert_equal(8, from_roman("ViIi"))
    end

    def test_from_roman_file
        file = IO.readlines("../test_data.txt")
        file.each do |line|
            integer, roman = line.strip.split(" ")
            integer = integer.to_i
            assert_equal(integer, from_roman(roman))
        end
    end
end
